### ISSUE TITLE
[Bug]: KryptonNumericUpDown loses selection (visually) each time the control is painted

### ISSUE THREAD

**Issue by Ahmed-Abdelhameed** [Krypton-Suite/Standard-Toolkit#39]
**Describe the bug**
Whenever the control is painted (i.e., when it receives a `WM_PAINT` message), the text selection gets lost (visually). Meaning that it's still selected but the selection isn't visible.

**To Reproduce**
1. Type something into a `KryptonNumericUpDown` control.
2. Select the text.
3. Do something that would cause the control to be repainted (e.g., move the cursor outside the control, minimize and restore the window, etc.)
4. See how the selection becomes invisible.

**Expected behavior**
The selection should remain visible as long as the control has focus.

**Screenshots**
![NUP](https://user-images.githubusercontent.com/18140178/102691937-cebbe200-4218-11eb-92ea-6cda588169e1.gif)

**Desktop:**
 - OS: Windows 10
 - Build: 2004.

**Additional context**
Okay, this is actually very related to #37. Let me elaborate...

At some point (when the `TextAlign` property was made hidden and `TextH` was added), the following part of the code in `WndProc` of the `SubclassEdit` class (handling the `WM_PAINT` message) was commented out and a comment was added:

```
//////////////////////////////////////////////////////
// Following removed to allow the Draw to always happen, to allow centering etc  
//// If enabled then let the combo draw the text area
//if (NumericUpDown.Enabled)
//{
//    // Let base implementation draw the actual text area
//    if (m.WParam == IntPtr.Zero)
//    {
//        m.WParam = hdc;
//        DefWndProc(ref m);
//        m.WParam = IntPtr.Zero;
//    }
//    else
//    {
//        DefWndProc(ref m);
//    }
//}
//else
```

My guess (and based on the comment above) is that it was commented out because it caused the `TextH` property not to take effect. However, I believe this has two problems:
1. It introduced this bug because custom-drawing the text doesn't take the selection into account, rendering it invisible.
2. It was the wrong fix, IMO. What should be done instead is what's suggested in #37, that is, either update the `TextAlign` property when `TextH` is changed or unhide `TextAlign` and encourage using it instead. Note that that's how `KryptonTextBox` works currently and its `TextH` doesn't even work. So, the easy fix would be to revert back to `TextAlign` and get rid of `TextH` if there are no plans to implement its logic properly (I'm not really sure it's even useful, to be honest).

TL;DR: if #37 gets fixed (using either of the suggested solutions), I believe the code above can be safely uncommented and it would fix this selection bug.

**Comment by Smurf-IV** [comment]
Enabling the code as suggested prevents the code from performing 
- Trailing Zero removal
- Centering Vertically
- Other fonts and alignments states dependent on focus / enablement

**Comment by Smurf-IV** [comment]
Sorted:
![NumericUpDowner](https://user-images.githubusercontent.com/2418812/155881952-63f944c1-1345-41c1-9418-d687880c3696.gif)
